sopel/modules/dice.py

```diff
@@ -132,6 +132,9 @@ def _roll_dice(bot, dice_expression):
         dice_expression,
         re.IGNORECASE | re.VERBOSE)
 
+    if result is None:
+        raise ValueError("Invalid dice expression: %r", dice_expression)
```

Comment:
As far as I can tell, exceptions don't handle string formatting for you the way logger calls do.

```pycon
>>> exc = None
>>> try:
...     raise ValueError("String: %r", "raw")
... except ValueError as e:
...     exc = e
...
>>> exc.args
('String: %r', 'raw')
>>> str(exc)
"('String: %r', 'raw')"
>>> print(exc)
('String: %r', 'raw')
```

```suggestion
        raise ValueError("Invalid dice expression: %r" % dice_expression)
```

Reply:
Oops, yea, that's just a mistake.